cmd/fs-v1.go

```diff
@@ -593,7 +594,7 @@ func (fs fsObjects) PutObject(bucket string, object string, data *hash.Reader, m
 		return ObjectInfo{}, traceError(errInvalidArgument)
 	}
 
-	metadata["etag"] = data.MD5HexString()
```

Comment:
IMO we still have to move this line below the the `fsCreate` Otherwise the etag is always the same for PUTs without content-md5

Reply:
Done @aead